src/main/java/seedu/contax/logic/commands/EditPriorityCommand.java

```diff
@@ -0,0 +1,133 @@
+package seedu.contax.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.contax.commons.core.Messages;
+import seedu.contax.commons.core.index.Index;
+import seedu.contax.logic.commands.exceptions.CommandException;
+import seedu.contax.model.Model;
+import seedu.contax.model.appointment.Appointment;
+import seedu.contax.model.appointment.Priority;
+
+
+/**
+ * Edits the priority level of an existing priority in the appointment list with level low, medium or high.
+ */
+public class EditPriorityCommand extends Command {
+    public static final String COMMAND_WORD = "priority";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits the level of priority of an existing appointment in the appointment list .\n"
+            + "Parameters: INDEX pri/PRIORITY\n"
+            + "Example: " + COMMAND_WORD + " 1 " + "pri/high";
+
+    public static final String MESSAGE_EDIT_PRIORITY_SUCCESS = "Edited Priority of Appointment at index %d: %s";
+    public static final String MESSAGE_REMOVE_PRIORITY_SUCCESS = "Removed Priority of Appointment at index %d";
+    public static final String MESSAGE_PRIORITY_NOT_EDITED =
+            "The priority level you provide is same as existing priority.";
+
+    private final Index index;
+    private final EditPriorityDescriptor editPriorityDescriptor;
+
+    /**
+     * Creates an {@code EditPriorityCommand} object.
+     *
+     * @param index The specified index to update the priority.
+     * @param editPriorityDescriptor Details to edit the priority with.
+     */
+    public EditPriorityCommand(Index index, EditPriorityDescriptor editPriorityDescriptor) {
+        requireNonNull(index);
+
+        this.index = index;
+        this.editPriorityDescriptor = editPriorityDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Appointment> appointmentList = model.getFilteredAppointmentList();
+
+        if (index.getZeroBased() >= appointmentList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_APPOINTMENT_DISPLAYED_INDEX);
+        }
+
+        Appointment appointment = appointmentList.get(index.getZeroBased());
+
+        if (editPriorityDescriptor.getPriority().isPresent()) {
+            Priority priority = editPriorityDescriptor.getPriority().get();
+
+            if (priority.equals(appointment.getPriority())) {
```

Comment:
Fairly certain `priority` here can be null, which will cause a `NullPointerException` when u call equals on it.

Reply:
priority is checked present in if condition, and appointment.getPriority() can be null and return false

Reply:
yah though effects are same can change to object.equals